### PR TITLE
feat: Add item_count on collection upsert

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR returns the `item_count` property when the collections are updated or inserted.

### CODE REVIEW

File: src/Collection/Collection.service.ts [hunk 1]
```diff
@@ -431,9 +431,9 @@ export class CollectionService {
     }
 
     const attributes = toDBCollection(collectionJSON)
-
     // Should we do something with the salt and the contract address? There's no need to have them
-    return new Collection(attributes).upsert()
+    // return new Collection(attributes).upsert()
```
Comment (juanmahidalgo) [inline src/Collection/Collection.service.ts:435]: Can we remove this comment here?

File: src/Collection/Collection.model.ts [hunk 2]
```diff
@@ -101,4 +101,35 @@ export class Collection extends Model<CollectionAttributes> {
...
+
+  static async upsertWithItemCount(
+    collection: CollectionAttributes
+  ): Promise<CollectionWithItemCount> {
+    const { id, ...attributes } = collection
+    const columnNames = ['id', ...Object.keys(attributes)]
+    const columnValues = [id, ...Object.values(attributes)]
+
+    const insertColumns = columnNames.map((name) => `"${name}"`)
+    const insertValues = columnNames.map((_, index) => `$${index + 1}`)
```
Comment (juanmahidalgo) [inline src/Collection/Collection.model.ts:113]: Can we use the `toAssignmentFields` method as we're doing [here](https://github.com/decentraland/builder-server/blob/54ba34e78b691a5ca789d0c5b8c7cdcb53430cf6/src/Curation/Curation.service.ts#L81)?

File: src/Collection/Collection.model.ts [hunk 3]
```diff
@@ -101,4 +101,35 @@ export class Collection extends Model<CollectionAttributes> {
...
+      .map((name, index) => `${name} = $${index + 2}`)
+
+    const result = await this.query<CollectionWithItemCount>(
+      SQL`
+    INSERT INTO ${raw(this.tableName)} as collections (${raw(
+        insertColumns.join(', ')
+      )})
+      VALUES (${raw(insertValues.join(', '))})
+      ON CONFLICT (id)
+      DO UPDATE SET ${raw(updateFields.join(', '))}
```
Comment (juanmahidalgo) [inline src/Collection/Collection.model.ts:126]: Could we pass directly the array `updateFields` so we avoid the `raw` and `.join(..`?

File: src/Collection/Collection.model.ts [hunk 4]
```diff
@@ -101,4 +101,35 @@ export class Collection extends Model<CollectionAttributes> {
...
+    collection: CollectionAttributes
+  ): Promise<CollectionWithItemCount> {
+    const { id, ...attributes } = collection
+    const columnNames = ['id', ...Object.keys(attributes)]
+    const columnValues = [id, ...Object.values(attributes)]
+
+    const insertColumns = columnNames.map((name) => `"${name}"`)
+    const insertValues = columnNames.map((_, index) => `$${index + 1}`)
+
+    const updateFields = columnNames
```
Comment (juanmahidalgo) [inline src/Collection/Collection.model.ts:115]: Can we use the `toAssignmentFields` method as we're doing [here](https://github.com/decentraland/builder-server/blob/54ba34e78b691a5ca789d0c5b8c7cdcb53430cf6/src/Curation/Curation.service.ts#L81)?

File: src/Collection/Collection.service.ts (hunk 1 above)
Comment (LautaroPetaccio) [reply src/Collection/Collection.service.ts]: Removed it!

File: src/Collection/Collection.model.ts (hunk 3 above)
Comment (LautaroPetaccio) [reply src/Collection/Collection.model.ts]: Changed it!

**Review by juanmahidalgo — APPROVED** [review]

**Pull request merged**